x/earn/keeper/grpc_query.go

```diff
@@ -266,8 +369,65 @@ func (s queryServer) getVaultAllDeposits(
 	}, nil
 }
 
-// getAccountAllDeposits returns deposits for all vaults for a specific account
-func (s queryServer) getAccountAllDeposits(
+// getBkavaVaultAllDeposits returns all deposits for the aggregated bkava vault
+func (s queryServer) getBkavaVaultAllDeposits(
+	ctx sdk.Context,
+	req *types.QueryDepositsRequest,
+) (*types.QueryDepositsResponse, error) {
+	deposits := []types.DepositResponse{}
+	store := prefix.NewStore(ctx.KVStore(s.keeper.key), types.VaultShareRecordKeyPrefix)
```

Comment:
will this have any performance concerns for these queries since we are iterating over all deposit shares every time?